lib/apiservers/engine/backends/container.go

```diff
@@ -108,6 +110,11 @@ var (
 
 	portMapper portmap.PortMapper
 
+	// bridge-to-bridge rules, indexed by mapped port;
+	// this map is used to delete the rule once
+	// the container stops or is removed
+	btbRules map[string][]string
```

Comment:
I see that we've got bunch of dictionaries which require synchronized access, which points that we need to use "generics" to generate typed code that will encapsulate and protect those dictionaries.